classes/Hook.php

```diff
@@ -1274,11 +1279,14 @@ private static function getMethodName(string $hookName): string
      */
     public static function getHookStatusByName($hook_name): bool
     {
-        $sql = new DbQuery();
-        $sql->select('active');
-        $sql->from('hook', 'h');
-        $sql->where('h.name = "' . pSQL($hook_name) . '"');
+        if (is_null(self::$active_hooks)) {
+            $sql = new DbQuery();
+            $sql->select('name');
+            $sql->from('hook', 'h');
+            $sql->where('h.active = 1');
+            self::$active_hooks = array_column(Db::getInstance()->executeS($sql), 'name');
```

Comment:
```suggestion
            static::$active_hooks = array_column(Db::getInstance()->executeS($sql), 'name');
```

Reply:
Oh one more suggestion, maybe you should only cache this if the result is not empty In case the method is called too early during installation for example